release_notes/ocp-4-10-release-notes.adoc

```diff
@@ -341,6 +341,13 @@ The Special Resource Operator (SRO) now exposes metrics to help you watch the he
 [id="ocp-4-10-dev-exp"]
 === Developer experience
 
+[id="ocp-4-10-dev-exp-prune-deployments"]
+==== Pruning deployment replica sets (Technology Preview)
```

Comment:
@soltysh I chatted with @deejross about the flag being experimental, so marking it as TP in the docs. Are you okay with this?

Reply:
Per slack convo, merging this PR. And if there are any concerns with this being marked TP, we can adjust in a future PR